### PR TITLE
SAGEConv

### PR DESCRIPTION
**Pull request by TristanBilot**
## Checklist

Put an `x` in the boxes that apply.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### CODE REVIEW

File: mlx_graphs/data/__init__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# from .data import GraphData # noqa
+# from .batch import (GraphDataBatch, batch, unbatch) # noqa
+# from .collate import collate # noqa
+
+# TODO: when uncommenting, this generates a circular import error, need tovfix that
+# to import GraphData with from mlx_graphs.data import GraphData
```
Comment (francescofarina) [inline mlx_graphs/data/__init__.py:6]: I don't see why that's happening. I just gave it a try and I don't have a circular import error when doing `from mlx_graphs.data import GraphData`.

File: mlx_graphs/nn/conv/sage_conv.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            neigh_features = self.propagate(
+                edge_index=edge_index, node_features=node_features, **message_kwargs
+            )
+            neigh_features = self.neigh_proj(neigh_features)
+
+        out_features = self.self_proj(node_features) + neigh_features
+
+        return out_features
+
+    def message(
```
Comment (francescofarina) [inline mlx_graphs/nn/conv/sage_conv.py:103]: I'm wondering whether we should make this method private - but I guess it's ok as it is

**Review by francescofarina — APPROVED** [review]
All good - just the think on circular imports

File: mlx_graphs/nn/conv/sage_conv.py (hunk 2 above)
Comment (TristanBilot) [reply mlx_graphs/nn/conv/sage_conv.py:103]: hmm, we have to think about that. It all depends more on the UX than on the good practices here

File: mlx_graphs/data/__init__.py (hunk 1 above)
Comment (francescofarina) [reply mlx_graphs/data/__init__.py:6]: FWIW - all tests run well on my end

Comment (TristanBilot) [reply mlx_graphs/data/__init__.py:6]: Did you change the occurrences: `from mlx_graphs.data.data import GraphData` to `from mlx_graphs.data import GraphData`?

Comment (francescofarina) [reply mlx_graphs/data/__init__.py:6]: I did, both in `test_data` and `test_batch`

**Pull request merged**